config/default.yaml

```diff
@@ -97,20 +97,33 @@ storage:
 
 object_storage:
   enabled: false
-  # Maximum amount of MiB to upload in one request to object storage
-  max_upload_part: 2048
-  # Will always use https with default URL generation (see below)
+
+  # Without protocol, will default to HTTPS
   endpoint: 's3.amazonaws.com'
+
+  region: 'us-east-1'
+
+  credentials:
+    access_key_id: 'access-key'
```

Comment:
I think it's a bad idea to encourage people to store secrets in a configuration file. The official s3 client supports loading from environment variables, which is more secure: https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html

Reply:
I'm not sure to understand why env variables are more secure since in the peertube context you would have to set them using a file to automatically load them no?

I agree we should find a way to set passwords for peertube in a more secure way (not only for object storage but also for smtp, redis, postgres...), but not sure what we could do.

I'll add a note about the env variable for object storage, I did not know this *auto load from env* thing